libraries/AP_BattMonitor/AP_BattMonitor.cpp

```diff
@@ -571,6 +571,33 @@ bool AP_BattMonitor::reset_remaining(uint16_t battery_mask, float percentage)
     return ret;
 }
 
+// Returns true if got mavlink charge state and fills charge_state The following mavlink charge states are not used
+// MAV_BATTERY_CHARGE_STATE_EMERGENCY , MAV_BATTERY_CHARGE_STATE_FAILED
+// MAV_BATTERY_CHARGE_STATE_UNHEALTHY, MAV_BATTERY_CHARGE_STATE_CHARGING, MAV_BATTERY_CHARGE_STATE_UNDEFINED
+bool AP_BattMonitor::get_mavlink_charge_state(const uint8_t instance, uint8_t &charge_state) const 
+{
+    if (instance >= _num_instances) {
+        return false;
+    }
+
+    switch (state[instance].failsafe) {
+
+    case AP_BattMonitor::Failsafe::None:
```

Comment:
```suggestion
    case Failsafe::None:
```

You're in the BattMonitor namespace so you don't need the extra crud.

Reply:
Opps! Fixed!